src/autocomplete_test.js

```diff
@@ -307,6 +318,49 @@ module.exports = {
 
         done();
     },
+    "test: liveAutocompleteDelay": function(done) {
+        var editor = initEditor("hello world ");
+        editor.setOptions({
+            liveAutocompletionDelay: 10,
+            liveAutocompletionThreshold: 2
+        });
+        
+        editor.completers = [{
+                getCompletions: function(editor, session, pos, prefix, callback) {
+                    callback(null,[{
+                        value: "test"
+                    }]);
+                }
+            },
+            {
+                getCompletions: function(editor, session, pos, prefix, callback) {
+                    this.timeout = setTimeout(() => {
+                        callback(null,[{
+                            value: "slow test"
+                        }]);
+                    }, 50);
+                },
+                cancel: function() {
+                    clearTimeout(this.timeout);
+                    this.timeout = null;
+                }
+            }
+        ];
+        
+        user.type("t");
+        user.type("e");
+        assert.ok(!editor.completer || !editor.completer.popup.isOpen);
+        setTimeout(function() {
+            assert.ok(editor.completer.popup.isOpen);
+            assert.ok(editor.completers[1].timeout);
```

Comment:
We might need to consider/revisit the slow completer case because we never get their results with live autocompletion when we set the delay. How about via keybinding? Do we need to wait for all completion results if triggered via keybinding?